specs/aenum.js

```diff
@@ -27,21 +27,25 @@ var SpecError = require('./error');
 module.exports.AEnum = AEnum;
 
 function AEnum(definitions) {
-    this.namesToValues = {};
-    this.valuesToNames = {};
+    this._namesToValues = {};
+    this._valuesToNames = {};
     var value = 0;
     for (var index = 0; index < definitions.length; index++) {
         var definition = definitions[index];
         var name = definition.id.name;
         value = definition.value != null ? definition.value : value;
-        if (this.namesToValues[name] !== undefined) {
+        if (this[name] !== undefined) {
+            throw new Error('Can\'t create enum with reserved name ' + name);
+        }
+        if (this._namesToValues[name] !== undefined) {
             throw new Error('Can\'t create enum with duplicate name ' + name + ' for value ' + value);
         }
         if (value > 0x7fffffff) {
             throw new Error('Can\'t create enum with value out of bounds ' + value + ' for name ' + name);
         }
-        this.namesToValues[name] = value;
-        this.valuesToNames[value] = name;
+        this._namesToValues[name] = value;
+        this._valuesToNames[value] = name;
+        this[name] = name;
```

Comment:
We really should not have dynamic fields on constructor fucntions; that defeats the point.

Constructors are not maps.